### PR TITLE
Rose bunch: bunch task counts at end of job

### PR DESCRIPTION
**Pull request by arjclark**
Addresses the first bullet point of #1876.

Adds a set of counts for job states to the end of the main job.out file. Useful when you're bunching large numbers of tasks to get an overall summary at a glance before digging around.

@matthewrmshin - please review 1
@oliver-sanders - please review 2

### CODE REVIEW

File: lib/python/rose/apps/rose_bunch.py [hunk 1]
```diff
@@ -265,13 +288,17 @@ def run(self, app_runner, conf_tree, opts, args, uuid, work_files):
...
+                notrun += 1
                 cmd = commands.pop(key).get_command()
                 app_runner.handle_event(NotRunEvent(key, cmd),
                                         prefix=self.PREFIX_NOTRUN)
 
         if self.dao:
             self.dao.close()
 
+        # Report summary data in job.out file
+        app_runner.handle_event(SummaryEvent(run_ok,run_fail,run_skip,notrun))
```
Comment (matthewrmshin) [inline lib/python/rose/apps/rose_bunch.py]: Space after commas. PEP8 test should pick this up?

Comment (arjclark) [inline lib/python/rose/apps/rose_bunch.py]: Yeah, my bad. Ran tests and then renamed some variables to be clearer after that.

**Comment by arjclark** [comment]
@matthewrmshin - fixed the pep8.

**Comment by matthewrmshin** [comment]
Logic looks OK 1. Do we need to add some docs?

**Comment by arjclark** [comment]
> Logic looks OK 1. Do we need to add some docs?

I don't think so, or at least not at this point - it's just an EOJ style end of job count up so I'd say its fairly self documenting.

File: t/rose-task-run/38-app-bunch-counts/app/cutoff/bin/echoer [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (oliver-sanders) [inline t/rose-task-run/38-app-bunch-counts/app/cutoff/bin/echoer]: Is this file needed?

Comment (arjclark) [inline t/rose-task-run/38-app-bunch-counts/app/cutoff/bin/echoer]: Nope, well spotted. In fact the whole bin/ directory under cutoff can be removed.

**Comment by arjclark** [comment]
@oliver-sanders - redundant/extra/unnecessary folder removed.

**Comment by oliver-sanders** [comment]
Looks OK 2.
